x/ccv/provider/keeper/relay.go

```diff
@@ -162,6 +162,13 @@ func (k Keeper) EndBlockVSU(ctx sdk.Context) {
 	}
 }
 
+// BlocksUntilNextEpoch returns the number of blocks until the next epoch starts
+// Returns 0 if the current block is the last block of the epoch
+func (k Keeper) BlocksUntilNextEpoch(ctx sdk.Context) int64 {
```

Comment:
It seems you could easily move the logic inside the query itself without having to create the `BlocksUntilNextEpoch` method. Was this added to make testing easier? Couldn't we test the logic from inside `grpc_query_test.go`?

Reply:
To make tests easier and to make the query logic and the relaying logic more closely connected. I dont see a big downside